File: app/src/main/java/com/song/deviceinfo/ui/battery/BatteryFragment.java
package com.song.deviceinfo.ui.battery;

import com.song.deviceinfo.ui.base.BaseAdapter;
import com.song.deviceinfo.ui.base.BaseViewModel;
import com.song.deviceinfo.ui.base.NormalAdapter;
import com.song.deviceinfo.ui.base.NormalFragment;

import androidx.core.util.Pair;
import androidx.lifecycle.ViewModelProviders;

public class BatteryFragment extends NormalFragment<Pair<String, String>> {

    @Override
    protected BaseAdapter crateAdapter() {
        return new NormalAdapter(getContext());
    }

    @Override
    protected BaseViewModel crateViewModel() {
        return ViewModelProviders.of(this).get(BatteryViewModel.class);
    }

}

File: app/src/main/java/com/song/deviceinfo/ui/battery/BatteryFragment.kt
package com.song.deviceinfo.ui.battery

import androidx.core.util.Pair
import androidx.lifecycle.ViewModelProviders
import com.song.deviceinfo.ui.base.BaseAdapter
import com.song.deviceinfo.ui.base.BaseViewModel
import com.song.deviceinfo.ui.base.NormalAdapter
import com.song.deviceinfo.ui.base.NormalFragment

/**
 * Created by chensongsong on 2020/5/27.
 */
class BatteryFragment : NormalFragment() {
    override fun createAdapter(): BaseAdapter<Pair<String, String>, *> {
        return NormalAdapter(requireContext())
    }
    
    override fun createViewModel(): BaseViewModel<Pair<String, String>> {
        return ViewModelProviders.of(this).get(BatteryViewModel::class.java)
    }
    
    override val normalInfo: List<Pair<String, String>>
        get() = (viewModel as BatteryViewModel).getBatteryInfo(requireContext())
}
